online_meetings/bluejeans_report.py

```diff
@@ -0,0 +1,186 @@
+# standard libraries
+import json
+import logging
+import os
+import sys
+import yaml
+
+# third-party libraries
+import pandas as pd
+import furl
+import time
+import dateparser
+import pytz
+from datetime import datetime, timedelta
+import glob
+
+# importing required modules
+from zipfile import ZipFile
+
+# Initialize settings and globals
+
+logger = logging.getLogger(__name__)
+
+# Default time to sleep when hitting a 429 limit
+DEFAULT_SLEEP_TIME = 10
+
+# read configurations
+try:
+    with open(os.path.join(os.path.dirname(__file__), '../config/env.json')) as env_file:
+        ENV = yaml.safe_load(env_file.read())
+except FileNotFoundError:
+    sys.exit(
+        'Configuration file could not be found; please add env.json to the config directory.')
+
+logging.basicConfig(level=ENV.get('LOG_LEVEL', 'DEBUG'))
+
+url = "https://api.bluejeans.com/oauth2/token?Client"
+
+payload = "{ \"grant_type\": \"client_credentials\", \"client_id\": \"Metrics_BJN\", \"client_secret\": \"" + ENV.get('BLUEJEANS_CLIENT_SECRET') + "\"}"
+logger.info(payload)
+headers = {
+  'Accept': 'application/json',
+  'Content-Type': 'application/json'
+}
+
+token_expiration_time = datetime.now()
+response = requests.request("POST", url, headers=headers, data=payload).text.encode('utf8')
+token_json = json.loads(response)
+token = token_json["access_token"]
+logger.info(token)
+## the expiration time in seconds
+expires_in_sec = token_json["expires_in"]
+## minus for one minute, for safty check
+expires_in_sec = expires_in_sec - 60
+token_expiration_time = datetime.now() + timedelta(seconds=expires_in_sec)
+logger.info(f"{token} expires on: {token_expiration_time}")
+
+# construct param string BLUEJEANS_MEETINGS_PARAMS
+
+params = ENV["BLUEJEANS_MEETINGS_PARAMS"]
+f = furl.furl('')
+f.args = params
+logger.info(f.url)
+
+bl_url = ENV["BLUEJEANS_MEETINGS_URL"]
+url = f'{bl_url}{f.url}'
+logger.info(url)
+
+payload = {}
+headers = {
+  'Accept': 'application/json',
+  'Content-Type': 'application/json',
+  'Authorization': f'Bearer {token}'
+}
+
+page_size = 1000
+total_records = 0
+
+total_df = pd.DataFrame()
+
+params = ENV["BLUEJEANS_MEETINGS_PARAMS"]
+
+# TODO: the earliest time is hardcoded now
+early_date = dateparser.parse(ENV.get('ZOOM_EARLIEST_FROM', '2020-03-31')).date()
+
+# Only use the date as a paramter
+# Loop until yesterday (this will go until now() -1)
+for i in range((datetime.now(pytz.timezone('America/Detroit')).date() - early_date).days):
+    param_date = early_date + timedelta(days=i)
+    start = datetime(param_date.year, param_date.month, param_date.day, 0, 0, 0)
+    end = datetime(param_date.year, param_date.month, param_date.day, 23, 59, 59)
+    logger.info("start")
+    logger.info(start.astimezone().isoformat())
+    logger.info("end")
+    logger.info(end.astimezone().isoformat())
+    
+    url = "https://indigo-api.bluejeans.com/v1/enterprise/" + str(ENV.get('BLUEJEANS_ENTERPRISE_ID')) + "/indigo/reportJob/meetings?filter=\
```

Comment:
You don't have to construct the URL like this, if you have the params as a dict you can pass it with the params argument to requests.request. I think this would probably be more readable if you were using [str.format](https://realpython.com/python-string-formatting/#2-new-style-string-formatting-strformat) if you didn't want to do this.

Reply:
Created as new issue